web_ui/ui_test.go

```diff
@@ -496,3 +503,261 @@ func TestServerHostRestart(t *testing.T) {
 		}
 	})
 }
+
+func TestApiToken(t *testing.T) {
+	route := gin.New()
+	route.POST("/api/v1.0/createToken", createApiToken)
+	route.DELETE("/api/v1.0/deleteApiToken/:id", deleteApiToken)
+	route.GET("/privilegedRoute", func(ctx *gin.Context) {
+		authOption := token.AuthOption{
+			Sources: []token.TokenSource{token.Header},
+			Issuers: []token.TokenIssuer{token.APITokenIssuer},
+			Scopes:  []token_scopes.TokenScope{token_scopes.Monitoring_Scrape},
+		}
+
+		status, ok, err := token.Verify(ctx, authOption)
+		if !ok {
+			ctx.JSON(status, server_structs.SimpleApiResp{
+				Status: server_structs.RespFailed,
+				Msg:    err.Error(),
+			})
+			return
+		}
+	})
+
+	ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
+	defer func() { require.NoError(t, egrp.Wait()) }()
+	defer cancel()
+
+	tempFile, err := os.CreateTemp("", "web-ui-passwd-api-token")
+	if err != nil {
+		fmt.Println("Failed to setup web-ui-passwd file")
+		os.Exit(1)
+	}
+	defer os.Remove(tempFile.Name())
+
+	dirName := t.TempDir()
+	server_utils.ResetTestState()
+	viper.Set("ConfigDir", dirName)
+	config.InitConfig()
+	viper.Set("Server.UIPasswordFile", tempFile.Name())
+	err = config.InitServer(ctx, server_structs.OriginType)
+	require.NoError(t, err)
+
+	content := "admin:password\n"
+	_, err = tempFile.WriteString(content)
+	assert.NoError(t, err, "Error writing to temp password file")
+
+	err = tempFile.Sync()
+	require.NoError(t, err)
+
+	//Configure UI
+	err = configureAuthDB()
+	assert.NoError(t, err)
```

Comment:
It might be worth taking a second look at which of these should be a `require` and which should be an `assert`. The `require` word means "if this doesn't pass, the test is doomed and there's no additional information I might get by running other functions", while the `assert` means "the test should fail if this doesn't pass, but there might be useful information about why the test is broken in other failed assertions".

I might be wrong, but it feels like failure to configure the auth DB is pretty fatal.